Implement user level `air.toml` support (#513)
posit-dev/air: docs/configuration.qmd

Air also supports walking up the directory tree from the project root.
For example, if you ran `air format` from within `~/packages/dplyr/R`, then Air would look "up" one directory and would find and use `~/packages/dplyr/air.toml`.

### User level configuration

We highly recommend that you use a project level `air.toml` file to ensure that all collaborators share the same settings, however Air also supports a user level `air.toml`.
This is a convenient place to set your personal defaults for any project that doesn't specify its own configuration.

Air looks for a user level configuration file at:

-   `~/.config/air/air.toml` on Linux and macOS.
    The environment variable `XDG_CONFIG_HOME` can be set as an alternative to `~/.config`, i.e. if set, Air will look in `{XDG_CONFIG_HOME}/air/air.toml`.

-   `%APPDATA%\air\air.toml` on Windows.
    The environment variable `APPDATA` can be set as an alternative to `%APPDATA%`, i.e. if set, Air will look in `{APPDATA}\air\air.toml`.

A user level `air.toml` takes precedence over IDE settings when [IDE settings synchronization](#configuration-settings-synchronization) is supported.

> Because user level `air.toml`s exist in a directory outside of your project's tree, any `exclude` patterns you provide must match at any depth.

Reviewer: ```suggestion
Because user level `air.toml`s exist in a directory outside of your project's tree, any `exclude` patterns you provide must be _unrooted_, i.e. they must match at any depth.
```

> Any pattern that begins with `**/` is allowed, as well as simple patterns such as `file.R` and `folder/`, which are internally treated as `**/file.R` and `**/folder/`.

Reviewer: ```suggestion
An unrooted pattern is one that begins with `**/`. In addition, simple patterns such as `file.R` and `folder/`, are internally treated as `**/file.R` and `**/folder/`.
```

> Patterns with a leading or interior `/` are not allowed, such as `/foo.R` and `R/foo.R`, as these imply `{root}/foo.R` and `{root}/R/foo.R`, which isn't applicable with a user level `air.toml`.

Reviewer: ```suggestion
On the other hand, patterns with a leading `/` are _rooted_ and not allowed, e.g. `/foo.R`. In addition, patterns with an interior `/` like `R/foo.R` are also considered rooted and treated as `/R/foo.R`.

The distinction between rooted and non-rooted paths is a bit subtle but was chosen for consistency with Git rules.
```
